### PR TITLE
Feat/add md ds to connectors

### PR DESCRIPTION
**Pull request by Cervangirard**
Add the possibility to store the datasources object inside a connectors :
Main changes are :
 - all functions to recreate datasources
 - datasource is an attribute of connectors obj
 - new class for nested connectors : nested_connectors
 - new logic to use the datasources from connect or recreate it if the user doesn't use connect

How to test :
- use connect with a custom yaml file and use the datasources fct
- use the connectors fct with custom connector fct and unse the datasources fct

### CODE REVIEW

**Review by vladimirobucina — COMMENTED** [review]
Functionalities look ok. I just think we need to stay consistent when definin imports from packages in documentation. I had few other comments. Also don't know if we should enforce parameter checking in internal functions (at least checkmate). All in all it looks good.

File: R/conts_datasources.R [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+#' including its name, package, parameters, and whether it's an R6 class constructor.
+#'
+#' @param func_string A character string representing the function call.
+#' @return A list with class "clean_fct_info" containing:
+#'   \item{function_name}{The name of the function or R6 class}
+#'   \item{parameters}{A list of parameters passed to the function}
+#'   \item{is_r6}{A boolean indicating whether it's an R6 class constructor}
+#'   \item{package_name}{The name of the package containing the function}
+#' @importFrom rlang parse_expr expr_text
+#' @importFrom purrr compact
```
[the remark this answers is not in the document]
Comment (Cervangirard) [reply R/conts_datasources.R:84]: fully agree :) I need to align with Claude ^^ more seriously, I will add rlang as an import of the package

File: R/connectors.R [hunk 2]
```diff
@@ -27,15 +27,33 @@
 #' @export
 connectors <- function(...) {
   x <- rlang::list2(...)
+
+  if (!is.null(x$atasources) & !inherits(x$datasources, "cnts_datasources")) {
```
[the remark this answers is not in the document]
Comment (Cervangirard) [reply R/connectors.R:31]: yep, will use [[ also

File: R/connectors.R [hunk 3]
```diff
@@ -55,3 +73,78 @@ print.connectors <- function(x, ...) {
...
+#' # Assume we have a 'my_connectors' object with a 'datasources' attribute
+#' my_connectors <- list()
+#' attr(my_connectors, "datasources") <- list(source1 = "data1", source2 = "data2")
+#'
+#' # Using the function
+#' result <- datasources(my_connectors)
+#' print(result)
+#'
+#' @export
+datasources <- function(connectors) {
```
[the remark this answers is not in the document]
Comment (Cervangirard) [reply R/connectors.R:126]: we can add a chek on the class, makes sense here because it is an exported fct

**Review by vladimirobucina — APPROVED** [review]
Now's good.

**Pull request merged**